server/Dockerfile

```diff
@@ -1,17 +1,79 @@
-FROM node:9.4.0-alpine
+#
+#	PagerMon image
+#
+FROM slocomptech/bi-node:12
 
-RUN apk add --no-cache sqlite
+#
+#	Arguments
+#
+ARG BUILD_DATE
+ARG NODE_ENV
+ARG NPM_ARGS
+ARG TAG
+ARG VCS_REF
+ARG VCS_SRC
+ARG VERSION
 
-ENV NODE_ENV production
+#
+#	Labels
+#	@see https://github.com/opencontainers/image-spec/blob/master/annotations.md
+#	@see https://semver.org/
+#
+LABEL org.opencontainers.image.authors="PagerMon Team" \
+			org.opencontainers.image.created=${BUILD_DATE} \
+			org.opencontainers.image.description="PagerMon server" \
+			org.opencontainers.image.documentation="https://github.com/pagermon/pagermon" \
+			org.opencontainers.image.revision=${VCS_REF} \
+			org.opencontainers.image.source=${VCS_SRC} \
+			org.opencontainers.image.title="PagerMon server" \
+			org.opencontainers.image.url="https://github.com/pagermon/pagermon"\
+			org.opencontainers.image.vendor="PagerMon Team" \
+			org.opencontainers.image.version=${VERSION}
 
-ADD . /app
+#
+#	Environment variables
+#	Don't forget to set NODE_ENV
+#
+ENV APP_NAME=PagerMon \
+    APP_VERSION=${VERSION} \
+		NODE_ENV=${NODE_ENV:-production}
 
-WORKDIR /app
+#
+#	Install app in container
+#
 
-RUN npm install pm2@latest -g && npm install
+#
+# s6 overlay scripts
+#	@see https://github.com/just-containers/s6-overlay
+#
+COPY root /
 
-VOLUME ["/data"]
+COPY . /app/
+
+# Fix file permissions
+RUN chown -R $CONTAINER_USER:$CONTAINER_USER /app && \
+		chown -R $CONTAINER_USER:$CONTAINER_USER /defaults 
 
 EXPOSE 3000
 
-CMD ["sh","/app/entrypoint.sh"]
+# Install pagermon
+RUN apk add --no-cache sqlite && \
+		apk add --no-cache --virtual .build-dependencies \
+			autoconf \
+			automake \
+			build-base \
+			cmake \
+			gcc \
+			g++ \
+			make \
+			musl-dev \
+			python3 && \
+		ln -s /usr/bin/python3 /usr/bin/python && \
+		cd /app && \
+		sudo -u ${CONTAINER_USER} -g ${CONTAINER_USER} npm install ${NPM_ARGS} && \
+		rm /usr/bin/python && \
```

Comment:
Why create symlink then delete? Is it needed for that npm install?

Reply:
npm build scripts require python for building libs from source, unfortunately when installing python3 there is no alias python created, which is needed, because python is specified in scripts like `#!/usr/bin/python`.

Reply:
Gotchya 👍